backend/src/controllers/process.js

```diff
@@ -12,33 +12,113 @@ async function processPdf(req, res) {
+
+                                const startIndex = index;
+                                const endIndex = startIndex + term.length;
+
+                                const contextBefore = extractedText.substring(
+                                    Math.max(startIndex - 100, 0),
+                                    startIndex
+                                );
+
+                                const contextAfter = extractedText.substring(
+                                    endIndex,
+                                    Math.min(endIndex + 200, extractedText.length)
+                                );
+
+                                let backLineIndex = contextBefore.lastIndexOf('\n')
+                                backLineIndex += 2
+                                const newLineindex = contextAfter.indexOf('\n')
+
+                                if (newLineindex && backLineIndex) {
+                                    let finalAfter = contextAfter.slice(0, newLineindex)
+                                    let finalBefore = contextBefore.slice(backLineIndex, startIndex)
+                                    let highlightedContent = `${term}: ${finalBefore}${term}${finalAfter}`
+                                    responseArray.push(highlightedContent)
+                                } else if (backLineIndex) {
+                                    let finalBefore = contextBefore.slice(backLineIndex, startIndex)
+                                    let highlightedContent = `${term}: ${finalBefore}${term}${contextAfter}`
+                                    responseArray.push(highlightedContent)
+                                } else if (newLineindex) {
+                                    let finalAfter = contextAfter.slice(0, newLineindex)
+                                    let highlightedContent = `${term}: ${contextBefore}${term}${finalAfter}`
+                                    responseArray.push(highlightedContent)
 
-                    const contextBefore = extractedText.substring(
-                        Math.max(startIndex - 100, 0),
-                        startIndex
-                    );
-                    const contextAfter = extractedText.substring(
-                        endIndex,
-                        Math.min(endIndex + 10, extractedText.length)
-                    );
+                                } else {
+                                    let highlightedContent = `${term}: ${contextBefore}${term}${contextAfter}`
+                                    responseArray.push(highlightedContent)
 
-                    const highlightedContent = `${searchTerm}${contextAfter}`;
+                                }
+                            })
+                        }
+                    })
+                } else if (findingParameter == 'first') {
+
+                    searchArray.forEach(term => {
+                        if (extractedText.includes(term)) {
```

Comment:
this logic should be in separate function. ideally same code should not be repeated again

Reply:
sir i will do this in future